projects/igniteui-angular/src/lib/core/utils.ts

```diff
@@ -576,6 +576,26 @@ export const yieldingLoop = (count: number, chunkSize: number, callback: (index:
 
 export const isConstructor = (ref: any) => typeof ref === 'function' && Boolean(ref.prototype) && Boolean(ref.prototype.constructor);
 
+/**
+ * Bind to the i18n manager's onResourceChange event
+ * @param destroyObj Object responsible for signaling destruction of the handling object
+ * @param context Reference to the object's this context
+ */
+export function onResourceChangeHandle(destroyObj: Subject<any> | DestroyRef, callback: (event?: CustomEvent<IResourceChangeEventArgs>) => void, context: any) {
+    const onResourceChangeHandler = callback.bind(context);
+    getI18nManager().addEventListener("onResourceChange", onResourceChangeHandler);
+
+    // Handle removal of listener on context destroy
+    const removeHandler = () => {
+        getI18nManager().removeEventListener("onResourceChange", onResourceChangeHandler);
+    }
+    if (destroyObj instanceof DestroyRef) {
+        destroyObj.onDestroy(() => removeHandler());
+    } else {
+        destroyObj.pipe(finalize(() => removeHandler()));
```

Comment:
I may be wrong, but I think this finalize() here will never be called.
When you .pipe() you're creating a new observable than no one will ever subscribe to. It will be GC as soon as this method finishes.

Reply:
I see it's resolved now.